### PR TITLE
PADV-1658 test: Shared Components Test Suite

### PR DESCRIPTION
**Pull request by nandodev-net**
## Tickets

- https://agile-jira.pearson.com/browse/PADV-1658

## Description
This PR focuses on enhancing and completing the test coverage for shared components utilized across the application. The aim is to ensure that all reusable components are thoroughly tested to guarantee their reliability and robustness.

Includes:

- **Shared Components:**

- AlertMessage: Verifies rendering, message display, and error handling logic.
- LabDetailsCard: Tests for correct lab detail representation, edge cases, and data handling.
- LabDetailsChartCard: Ensures that chart data is correctly processed and displayed.
- Table: Covers table rendering, data population, empty states, and interactions like sorting and pagination.
- TableFilter: Validates filtering logic, form handling, and integration with the table component.
- DashboardLaunchButton: Ensures button visibility, click events, error handling, and integration with the dashboard launch logic.
- JsonViewer: Verifies JSON editor initialization, configuration, and cleanup behavior.

## Changes Made

This PR aims to achieve full coverage for all shared components, ensuring they function correctly in all intended use cases. Expected coverage for all shared components is at least 87%

## How To Test

1. Ensure your development environment is set up with the latest dependencies by running `npm i` in your MFE folder.
2. Run all test with `npm test -- --verbose`
Verify that all tests pass and that coverage reports indicate the coverage for the targeted files.

### Component Tests

- **Shared**

    - [x] AlertMessage ----> **Done** cov. 100/100
    - [x] LabDetailsCard ----> **Done** cov. 100/100
    - [x] LabDetailsChatCard ----> **Done** cov. 100/100
    - [x] Table ----> **Done** cov. 100/100
    - [x] TableFilter ----> **Done** cov. 90.47/100
    - [x] DashboardLaunchButton  ---> **Done** cov. 91.66/100
    - [x] JsonViewer  ---> **Done** cov. 100/100
    
    
image

### CODE REVIEW

File: src/shared/AlertMessage/tests/index.test.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import AlertMessage from '../index';
+
+describe('AlertMessage Component', () => {
+  it('renders with default props', () => {
+    render(<AlertMessage heading="Test Heading" />);
+
+    // Check that the heading is rendered
```
Comment (01001110J) [inline src/shared/AlertMessage/tests/index.test.jsx:10]: I think this comments are not necessary

File: src/shared/AlertMessage/tests/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  });
+
+  it('renders children correctly', () => {
+    render(
+      <AlertMessage heading="Test Heading">
+        <button type="button">Click me</button>
+      </AlertMessage>,
+    );
+
+    // Check that the child element is rendered
```
Comment (01001110J) [inline src/shared/AlertMessage/tests/index.test.jsx:39]: Same for this comment

File: src/shared/DashboardLaunchButton/index.jsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { Button } from 'react-paragon-topaz';
 
 import AlertMessage from 'shared/AlertMessage';
 import { skillableUrl, defaultErrorMessage } from 'constants';
-import { eventManager } from 'helpers';
+import { eventManager } from '../../helpers';
```
Comment (01001110J) [inline src/shared/DashboardLaunchButton/index.jsx:9]: Is not possible to use a alias path?

File: src/shared/DashboardLaunchButton/tests/index.test.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  render,
+  screen,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import DashboardLaunchButton from '../index';
+import '@testing-library/jest-dom';
+
+// Mock necessary dependencies
```
Comment (01001110J) [inline src/shared/DashboardLaunchButton/tests/index.test.jsx:12]: Unnecessary comment

File: src/shared/JsonViewer/tests/index.test.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  it('initializes JSONEditor with correct options and data', () => {
+    render(<JsonViewer labName={mockLabName} labData={mockLabData} />);
+
+    // Verify that the JSONEditor was initialized properly
+    // You can add checks to see if the lab data is displayed in the viewer
+    const editorContainer = document.querySelector('.jsoneditor');
+    expect(editorContainer).toBeInTheDocument();
+
+    // More checks can be done here depending on the behavior of the library
```
Comment (01001110J) [inline src/shared/JsonViewer/tests/index.test.jsx:29]: image

File: src/shared/DashboardLaunchButton/tests/index.test.jsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    mockPost.mockResolvedValueOnce({ data: { is_ccx_course: true } });
+    mockPost.mockResolvedValueOnce({ data: { url: 'https://example.com/dashboard' } });
+
+    render(<DashboardLaunchButton courseId={mockCourseId} title="Test Title" />);
+
+    await waitFor(() => {
+      expect(screen.getByRole('button', { name: /Go To Instructor Dashboard/i })).toBeInTheDocument();
+    });
+
+    window.open = jest.fn(); // Mock window.open
```
Comment (01001110J) [inline src/shared/DashboardLaunchButton/tests/index.test.jsx:65]: Unnecessary comment

File: src/shared/JsonViewer/tests/index.test.jsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  });
+
+  it('destroys JSONEditor instance on component unmount', () => {
+    const { unmount } = render(<JsonViewer labName={mockLabName} labData={mockLabData} />);
+
+    // Unmount the component and ensure no errors occur
```
Comment (01001110J) [inline src/shared/JsonViewer/tests/index.test.jsx:35]: Unnecessary comment

File: src/shared/LabDetailsCard/tests/index.test.jsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import LabDetailsCard from '../index';
+
+describe('LabDetailsCard Component', () => {
+  it('renders with default props', () => {
+    render(<LabDetailsCard />);
+
+    // Check that all default fields are rendered correctly
```
Comment (01001110J) [inline src/shared/LabDetailsCard/tests/index.test.jsx:10]: Unnecessary comment

File: src/shared/LabDetailsChartCard/tests/index.test.jsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+    // Check for the presence of the Gauge charts by role
+    const gauges = screen.getAllByRole('meter'); // 'meter' is the role for gauge elements
+    expect(gauges).toHaveLength(2);
+  });
+
+  it('handles missing details correctly', () => {
+    const partialDetails = {
+      NumTasks: 5,
+      // Missing other fields
```
Comment (01001110J) [inline src/shared/LabDetailsChartCard/tests/index.test.jsx:62]: ?

File: src/shared/LabDetailsChartCard/tests/index.test.jsx [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    const completedTasksElements = screen.getAllByText('COMPLETED TASKS');
+    const completedTasksItem = completedTasksElements[0].closest('.lab-details-item');
+    const completedTasksSpan = completedTasksItem.querySelector('span');
+
+    const defaultGauges = screen.getAllByRole('meter'); // Get all gauges
+
+    expect(completedTasksItem).not.toBeNull();
+    expect(completedTasksSpan).not.toBeNull();
+    expect(completedTasksSpan).toHaveTextContent('');
+    expect(defaultGauges).toHaveLength(2); // Expecting 2 gauges
```
Comment (01001110J) [inline src/shared/LabDetailsChartCard/tests/index.test.jsx:80]: Unnecessary comment

File: src/shared/LabDetailsChartCard/tests/index.test.jsx [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    expect(defaultGauges[0].hasAttribute('aria-valuenow')).toBe(false); // Expecting the attribute to be missing
```
Comment (01001110J) [inline src/shared/LabDetailsChartCard/tests/index.test.jsx:81]: Unnecessary comment

File: src/shared/Table/tests/index.test.jsx [hunk 12]
```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Table from '../index';
+
+// Mock props
```
Comment (01001110J) [inline src/shared/Table/tests/index.test.jsx:6]: Unnecessary comment

**Review by 01001110J — COMMENTED** [review]
In general I think the 98% of the comments are not necessary and that's because the test is descriptive itself maybe in some cases like the gauges make sense.

File: src/shared/DashboardLaunchButton/index.jsx (hunk 3 above)
Comment (nandodev-net) [reply src/shared/DashboardLaunchButton/index.jsx:9]: when i test the alias path explodes for some reason

**Comment by nandodev-net** [comment]
> In general I think the 98% of the comments are not necessary and that's because the test is descriptive itself maybe in some cases like the gauges make sense.

Sure!, Sure, but given so many tests of that tool that I don't know well, for me they were very useful. I Will erased all those comments

File: src/shared/LabDetailsChartCard/tests/index.test.jsx (hunk 10 above)
Comment (01001110J) [reply src/shared/LabDetailsChartCard/tests/index.test.jsx:74]: @nandodev-net

**Review by 01001110J — APPROVED** [review]

**Review by JuanDavidBuitrago — APPROVED** [review]
